src/compiler/checker.ts

```diff
@@ -14628,16 +14604,18 @@ namespace ts {
             if (!ignoreReturnTypes) {
                 const sourceTypePredicate = getTypePredicateOfSignature(source);
                 const targetTypePredicate = getTypePredicateOfSignature(target);
-                result &= sourceTypePredicate !== undefined || targetTypePredicate !== undefined
-                    ? compareTypePredicatesIdentical(sourceTypePredicate, targetTypePredicate, compareTypes)
-                    // If they're both type predicates their return types will both be `boolean`, so no need to compare those.
-                    : compareTypes(getReturnTypeOfSignature(source), getReturnTypeOfSignature(target));
+                result &= sourceTypePredicate || targetTypePredicate ?
+                    compareTypePredicatesIdentical(sourceTypePredicate, targetTypePredicate, compareTypes) :

```

Comment:
I see the team is split on ternary style (trailing punctuation vs leading). Should we enforce one via linter?

Reply:
In my original implementation of the compiler it was all consistently trailing punctuation and it still is in the majority of the code base. I'd prefer to just standardize on that.

Reply:
I generally prefer to have trailing punctuation when there are multiple cases (similar to a switch), but leading punctuation when there is a single condition (like an if/else).